riposte-core/src/main/java/com/nike/riposte/util/AsyncNettyHelper.java

```diff
@@ -505,4 +509,80 @@ public static boolean executeOnlyIfChannelIsActive(ChannelHandlerContext ctx, St
             }
         }
     }
+
+    /**
+     * Helper method for creating a `CompletableFuture` that is using the tracing helpers.
+     * <p>
+     * <pre>
+     * AsyncNettyHelper.executeAsyncCall(() -> {
+     *      //do some work in a background thread
+     *      return VOID;
+     * }, executor, ctx);
+     * </pre>
+     */
+    public static <U> CompletableFuture<U> executeAsyncCall(Supplier<U> f, Executor executor, ChannelHandlerContext ctx) {
+        return CompletableFuture.supplyAsync(supplierWithTracingAndMdc(f, ctx), executor);
+    }
+
+    /**
+     * Helper method for creating a `CompletableFuture` that is using the tracing helpers and has the CompletableFuture
+     * wrapped in a `CircuitBreaker`
+     * <p>
+     * <pre>
+     * AsyncNettyHelper.executeAsyncCall(() -> {
+     *      //do some work in a background thread
+     *      return VOID;
+     * }, circuitBreaker, executor, ctx);
+     * </pre>
+     */
+    public static <U> CompletableFuture<U> executeAsyncCall(Supplier<U> f, CircuitBreaker<U> circuitBreaker, Executor executor, ChannelHandlerContext ctx) {
+        return circuitBreaker.executeAsyncCall(() -> executeAsyncCall(f, executor, ctx));
+    }
+
+    /**
+     * Helper method for creating a `CompletableFuture` that is wrapped by a CircuitBreaker and
+     * has the `Supplier` wrapped around a SubSpan.
+     * <p>
+     * You would prefer this method over the above methods when your `Supplier` has logic that makes an outbound/downstream call.
```

Comment:
Rather than `You would prefer this method over the above ...`, how about `You would prefer this method over {@link #theOtherMethod()} ...` so readers aren't wondering which method you're referring to when you say "the above"?

Reply:
good call. i've added the links